tangelo/algorithms/projective/tests/test_qpe.py

```diff
@@ -0,0 +1,121 @@
+        self.assertEqual(resources["circuit_width"], 10)
+        self.assertEqual(resources["circuit_2qubit_gates"], 20464)
+        self.assertEqual(resources["circuit_depth"], 28110)
+
+    def test_simulate_h2_circuit(self):
+        """Run QPE on H2 molecule, with JW qubit mapping and exact simulator providing only the Trotter circuit
+        """
+
+        ref_circ = get_reference_circuit(mol_H2_sto3g.n_active_sos, mol_H2_sto3g.n_active_electrons, "scbk", True, 0)
+        qu_op = fermion_to_qubit_mapping(mol_H2_sto3g.fermionic_hamiltonian, "scbk", mol_H2_sto3g.n_active_sos,
+                                         mol_H2_sto3g.n_active_electrons, True, 0)
+        unit_circ = trotterize(mol_H2_sto3g.fermionic_hamiltonian, 2*np.pi, 1, 4, True,
+                               {"qubit_mapping": "scbk", "up_then_down": True, "n_spinorbitals": mol_H2_sto3g.n_active_sos,
+                                "n_electrons": mol_H2_sto3g.n_active_electrons})
+
+        # Test supplying circuit and applying QPE controls to only gates marked as variational
+        qpe_options = {"qubit_hamiltonian": qu_op, "unitary": unit_circ, "size_qpe_register": 8, "ref_state": ref_circ,
+                       "backend_options": {"target": "qulacs"}, "unitary_options": {"control_method": "variational"}}
+        qpe_solver = QPESolver(qpe_options)
+        qpe_solver.build()
+
+        energy = qpe_solver.simulate()
+
+        self.assertAlmostEqual(energy, -(-1.13727-qu_op.constant), delta=1e-3)
+
+        # Test supplying circuit with QPE controls added to every gate.
+        qpe_options = {"qubit_hamiltonian": qu_op, "unitary": unit_circ, "size_qpe_register": 8, "ref_state": ref_circ,
+                       "backend_options": {"target": "qulacs"}, "unitary_options": {"control_method": "all"}}
+        qpe_solver = QPESolver(qpe_options)
+        qpe_solver.build()
+
+        energy = qpe_solver.simulate()
+
+        self.assertAlmostEqual(energy, -(-1.13727-qu_op.constant), delta=1e-3)
+
+    def test_qubit_hamiltonian_input(self):
+        """Test with qubit hamiltonian input."""
+
+        # Generate qubit operator with state 9 having eigenvalue 0.25
+        qu_op = (QubitOperator("X0 X1", 0.125) + QubitOperator("Y1 Y2", 0.125) + QubitOperator("Z2 Z3", 0.125)
+                 + QubitOperator("", 0.125))
+
+        ham_mat = get_sparse_operator(qu_op.to_openfermion()).toarray()
+        _, wavefunction = np.linalg.eigh(ham_mat)
+
+        sv = StateVector(wavefunction[:, 9], order="lsq_first")
+        init_circ = sv.initializing_circuit()
+
+        qpe = QPESolver({"qubit_hamiltonian": qu_op, "size_qpe_register": 4, "ref_state": init_circ, "backend_options": {"target": "cirq"},
+                         "unitary_options": {"time": -2*np.pi, "n_trotter_steps": 1, "n_steps_method": "repeat", "trotter_order": 4}})
+        qpe.build()
+        energy = qpe.simulate()
+        self.assertAlmostEqual(energy, 0.25, delta=1.e-10)
```

Comment:
This delta is too small, too close to machine precision. Reordering of mathematical operations, rounding errors or tiny update in underlying computing libraries could break that. Is there a reason we are going that far ?

Is 0.250001 for instance already not good enough ? Can't we settle for 1e-5 or something ?